### PR TITLE
Fix typos in Enum documentation

### PR DESCRIPTION
**Pull request by pgeraghty**

### CODE REVIEW

File: lib/elixir/lib/enum.ex [hunk 1]
```diff
@@ -162,7 +162,7 @@ defprotocol Enumerable do
   Otherwise it should return `{:error, __MODULE__}` and a default algorithm
   built on top of `reduce/3` that runs in linear time will be used.
 
-  When not called whithin guards, the [`in`](`in/2`) and [`not in`](`in/2`)
+  When not called within guards, the [`in`](`in/2`) and [`not in`](`in/2`)
```
Comment (eksperimental) [inline lib/elixir/lib/enum.ex:165]: Wouldn't it be clearer to say "When called outside guards, " ?

**Review by eksperimental — COMMENTED** [review]
suggestion

File: lib/elixir/lib/enum.ex [hunk 2]
```diff
@@ -1681,7 +1681,7 @@ defmodule Enum do
       false
 
 
-  When not called whithin guards, the [`in`](`in/2`) and [`not in`](`in/2`)
+  When not called within guards, the [`in`](`in/2`) and [`not in`](`in/2`)
```
Comment (josevalim) [inline lib/elixir/lib/enum.ex:1684]:
```suggestion
  When called outside guards, the [`in`](`in/2`) and [`not in`](`in/2`)
```

File: lib/elixir/lib/enum.ex (hunk 1 above)
Comment (josevalim) [inline lib/elixir/lib/enum.ex:165]:
```suggestion
  When called outside guards, the [`in`](`in/2`) and [`not in`](`in/2`)
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**